### PR TITLE
Fix broken hack/verify-symbols.sh

### PR DESCRIPTION
**Pull request by sttts**
The operating system nm does not output anything anymore for our go binaries. `go tool nm` does.

Fixing #77983

### CODE REVIEW

**Review by nikhita — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
/approve

looks like verify still fails @sttts 

```
reading /home/prow/go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/hyperkube: no symbol section
reading /home/prow/go/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/hyperkube: no symbols
```

**Comment by sttts** [comment]
On mac with `make WHAT=cmd/hyperkube` nm doesn't show symbols, but `go tool build`  does. But in the tests not even the latter does.

**Comment by nikhita** [comment]
I'm on Linux..

`make what=cmd/hyperkube` will strip symbols, so `nm` (and `go tool nm`) doesn't work. Only `nm -D` will work, but it'll show dynamic symbols which aren't of use to us.

```
➜  kubernetes git:(master) ✗ file _output/local/bin/linux/amd64/hyperkube
_output/local/bin/linux/amd64/hyperkube: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=96bc7b98c84ba74bec6be0d67a7338518098f6a2, stripped
```

> but go tool build does. 

Did you mean `go build`? I tried go build and it does _not_ strip symbols, so `nm` actually ends up working there.

File: hack/verify-symbols.sh [hunk 1]
```diff
@@ -30,14 +30,15 @@ BADSYMBOLS=(
   "httptest"
   "testify"
   "testing[.]"
+  "TrackStorageCleanup"
 )
 
 # b/c hyperkube binds everything simply check that for bad symbols
-SYMBOLS="$(nm "${KUBE_OUTPUT_HOSTBIN}/hyperkube")"
```
Comment (jpbetz) [inline hack/verify-symbols.sh]:
```sh
GOLDFLAGS="-w" make -C "${KUBE_ROOT}" WHAT=cmd/hyperkube
```

Might do the trick. Looks like `GOLDFLAGS` defaults to `-s -w` in `kubernetes/hack/lib/golang.sh` where the `-s` strips out symbols.

The only problem with this is that we leave a `_output/local/bin/linux/amd64/hyperkube` binary in the project that doesn't have symbols stripped. If we can build a custom output directory that might be safer..

**Comment by fejta** [comment]
Looks like pull-kubernetes-verify is failing. Can you try and fix this?

File: hack/verify-symbols.sh (hunk 1 above)
Comment (sttts) [reply hack/verify-symbols.sh]: fixed

**Comment by jpbetz** [comment]
Thanks @sttts!

/lgtm

**Pull request merged**